src/PrestaShopBundle/Bridge/AdminController/FrameworkBridgeControllerTrait.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Bridge\AdminController;
+
+use PrestaShopBundle\Bridge\Exception\BridgeException;
+use Tab;
+
+/**
+ * Contains reusable methods for horizontally migrated controllers
+ */
+trait FrameworkBridgeControllerTrait
+{
+    /**
+     * @var LegacyControllerBridgeInterface|null
+     */
+    private $legacyControllerBridge;
+
+    /**
+     * @var ControllerConfiguration|null
+     */
+    private $controllerConfiguration;
+
+    /**
+     * @return ControllerConfiguration
+     */
+    public function getControllerConfiguration(): ControllerConfiguration
+    {
+        if (!$this->controllerConfiguration) {
```

Comment:
I don't think this should result in an error, with your current implementation you are implicitly forcing `buildLegacyControllerBridge` (and therefore indirectly `getLegacyControllerBridge()`) to be called FIRST so that it build the two objects and only after can you call `getControllerConfiguration` safely

Each getters should be callable independently in any order, meaning that this piece of code:
```php
$tabId = Tab::getIdFromClassName($legacyControllerName);

        if (!$tabId) {
            throw new BridgeException(sprintf(
                'Tab not found by className "%s". Make sure that $legacyControllerName is correct',
                $legacyControllerName
            ));
        }

        $this->controllerConfiguration = $this
            ->get('prestashop.bridge.admin_controller.controller_configuration_factory')
            ->create($tabId, $objectModelClassName, $legacyControllerName, $tableName)
        ;
```

should be in this getter, and in the `buildLegacyControllerBridge` you can then call `getControllerConfiguration()` to build the bridge.